### PR TITLE
Validate the input field when leaving the field

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6250/
Test PASSed.

**Comment by anthonydillon** [comment]
## QA
- Hook up JIMM in `development.ini`
- Deploy an application
- Begin to enter an incorrect name for a model (such as, "-abc-") 
- See that the validation does not appear until you click off the input
- Check the input to something validate (such as "abc") Click off and see validation passes
- Do the same for the credentials name
- Leave an empty input and you should now only get the "This is required" validation instead of the character check too

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js [hunk 1]
```diff
@@ -277,7 +277,7 @@ YUI.add('deployment-credential-add', function() {
                   regex: /\S+/,
                   error: 'This field is required.'
                 }, {
-                  regex: /^[a-zA-Z0-9]([a-zA-Z0-9.-]*[a-zA-Z0-9])?$/,
+                  regex: /^([a-zA-Z0-9]([a-zA-Z0-9.-]*[a-zA-Z0-9])?)?$/,
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js]: IIRC this regex was grabbed from the Juju source - Can you confirm that this is still correct?

**Comment by anthonydillon** [comment]
By adding the brackets it means the regex no passes on empty string. The
validation above it catches empty string.

**Comment by urosj** [comment]
QA 🆗

**Comment by urosj** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**